internal/operators/ocs/validations.go

```diff
@@ -2,247 +2,251 @@ package ocs
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/docker/go-units"
 	"github.com/openshift/assisted-service/internal/operators/api"
 	"github.com/openshift/assisted-service/models"
-	"github.com/sirupsen/logrus"
 )
 
-//go:generate mockgen -source=validations.go -package=ocs -destination=mock_validations.go
-type OCSValidator interface {
-	ValidateRequirements(cluster *models.Cluster) (api.ValidationStatus, string)
-}
-
-type ocsValidator struct {
-	*Config
-	log logrus.FieldLogger
-}
-
-func NewOCSValidator(log logrus.FieldLogger, cfg *Config) OCSValidator {
-	return &ocsValidator{
-		log:    log,
-		Config: cfg,
-	}
+type Config struct {
+	OCSMinimumCPUCount             int64             `envconfig:"OCS_MINIMUM_CPU_COUNT" default:"18"`
+	OCSMinimumRAMGB                int64             `envconfig:"OCS_MINIMUM_RAM_GB" default:"51"` // disable deployment if less
+	OCSRequiredDisk                int64             `envconfig:"OCS_MINIMUM_DISK" default:"3"`
+	OCSRequiredDiskCPUCount        int64             `envconfig:"OCS_REQUIRED_DISK_CPU_COUNT" default:"2"` // each disk requires 2 cpus
+	OCSRequiredDiskRAMGB           int64             `envconfig:"OCS_REQUIRED_DISK_RAM_GB" default:"5"`    // each disk requires 5GB ram
+	OCSRequiredHosts               int64             `envconfig:"OCS_MINIMUM_HOST" default:"3"`
+	OCSRequiredCPUCount            int64             `envconfig:"OCS_REQUIRED_CPU_COUNT" default:"24"`              // Standard Mode 8*3
+	OCSRequiredRAMGB               int64             `envconfig:"OCS_REQUIRED_RAM_GB" default:"57"`                 // Standard Mode
+	OCSRequiredCompactModeCPUCount int64             `envconfig:"OCS_REQUIRED_COMAPCT_MODE_CPU_COUNT" default:"30"` // Compact Mode cpu requirements for 3 nodes( 4*3(master)+6*3(OCS) cpus)
+	OCSRequiredCompactModeRAMGB    int64             `envconfig:"OCS_REQUIRED_COMAPCT_MODE_RAM_GB" default:"81"`    //Compact Mode ram requirements (8*3(master)+57(OCS) GB)
+	OCSMasterWorkerHosts           int64             `envconfig:"OCS_REQUIRED_MASTER_WORKER_HOSTS" default:"6"`
+	OCSMinimalDeployment           bool              `envconfig:"OCS_MINIMAL_DEPLOYMENT" default:"false"`
```

Comment:
Just a small change I was thinking. `OCSMinimalDeployment` should be removed.

Reply:
In the manifest generation, we are using the `OCSDeploymentType` to check for different modes. Hence it is not used anywhere else.

Reply:
we are using that in manifest.go

Reply:
could you point me to where `OCSMinimalDeployment` is used in manifest.go ?

Reply:
Sorry I read it wrong, I removed it now

Reply:
Thanks!